### PR TITLE
LANG-1135 - Add method containsAllWords to WordUtils

### PR DESCRIPTION
**Pull request by eduardomartins-luizalabs**
Checks if the String contains all words.
```java
StringUtils.containsAllWords("lorem ipsum dolor sit amet", "ipsum", "lorem", "dolor") == true
StringUtils.containsAllWords("lorem ipsum dolor sit amet", "ipsum", null, "lorem", "dolor") == false
StringUtils.containsAllWords("ab", "b") == false
```

### CODE REVIEW

File: src/main/java/org/apache/commons/lang3/StringUtils.java [hunk 1]
```diff
@@ -1685,7 +1685,7 @@ public static boolean containsAny(final CharSequence cs, final CharSequence sear
      * StringUtils.containsAny("", *)              = false
      * StringUtils.containsAny(*, null)            = false
      * StringUtils.containsAny(*, [])              = false
-     * StringUtils.containsAny("abcd", "ab", "cd") = false
+     * StringUtils.containsAny("abcd", "ab", "cd") = true
```
Comment (agrava-luizalabs) [inline src/main/java/org/apache/commons/lang3/StringUtils.java]: It is good to point out that the documentation was wrong in this case

**Comment by britter** [comment]
This would better fit into ```org.apache.commons.lang3.text.WordUtils```, because it deals with words. I could imagine having a StringUtils.containsAll(CharSequence, CharSequence...) method that just checks if all of the given CharSequences are all contained in the first CharSequence, but without handling word boundaries.

**Comment by eduardomartins-luizalabs** [comment]
@britter i didn't saw this class, my bad.
Should I send another commit or amend this?

**Comment by britter** [comment]
Usually we make additional commit so that the development of a feature can be tracked afterwards. Since in this case the commit would change the whole PR, I think amending is appropriate in this case.

**Comment by eduardomartins-luizalabs** [comment]
Ok

File: src/test/java/org/apache/commons/lang3/StringUtilsEqualsIndexOfTest.java [hunk 2]
```diff
@@ -16,21 +16,16 @@
...
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.assertTrue;
-
-import java.util.Locale;
-
 import org.apache.commons.lang3.test.DefaultLocale;
 import org.hamcrest.core.IsNot;
 import org.junit.Test;
 
+import java.util.Locale;
```
Comment (eduardomartins-luizalabs) [inline src/test/java/org/apache/commons/lang3/StringUtilsEqualsIndexOfTest.java]: Sorry about that, Ide issue.
I will fix.

**Comment by eduardomartins-luizalabs** [comment]
@britter updated.

**Comment by britter** [comment]
Your placeholder Jira issue is https://issues.apache.org/jira/browse/LANG-1135. Please create a Jira for bigger contributions as documented in CONTRIBUTING.md. Thank you!

**Comment by britter** [comment]
Hello @eduardomartins-luizalabs,

the following will fail:

```java
assertFalse(WordUtils.containsAllWords("lorem ipsum null dolor sit amet", "ipsum", null, "lorem", "dolor"));
```

This is because you only check for nulls at index 0 of the input array. IMHO null is not contained in any string. Have a look at for example ```StringUtils.containsAny(CharSequence, CharSequence...)``` and adc6bc4104514063b3e8ebe7a301d5dd4bce36d3. Further more I wonder whether we should let this method operate on CharSequences. WDYT?

**Comment by eduardomartins-luizalabs** [comment]
Sorry about the JIRA card.
I think it's a great improvement, i will do that.

**Comment by eduardomartins-luizalabs** [comment]
@britter updated.

**Pull request merged**